feature/platform/transceiver/tests/zr_fec_uncorrectable_frames_test/zr_fec_uncorrectable_frames_test.go

```diff
@@ -38,6 +40,42 @@ func TestMain(m *testing.M) {
 	fptest.RunTests(m)
 }
 
+func opticalChannelComponentFromPort(t *testing.T, dut *ondatra.DUTDevice, p *ondatra.Port) string {
+	t.Helper()
+	if deviations.MissingPortToOpticalChannelMapping(dut) {
+		transceiverName := gnmi.Get(t, dut, gnmi.OC().Interface(p.Name()).Transceiver().State())
+		return fmt.Sprintf("%s-Optical0", transceiverName)
+	}
+	compName := gnmi.Get(t, dut, gnmi.OC().Interface(p.Name()).HardwarePort().State())
+	for {
+		comp, ok := gnmi.Lookup(t, dut, gnmi.OC().Component(compName).State()).Val()
+		if !ok {
+			t.Fatalf("Recursive optical channel lookup failed for port: %s, component %s not found.", p.Name(), compName)
+		}
+		if comp.GetType() == oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_OPTICAL_CHANNEL {
+			return compName
+		}
+		if comp.GetParent() == "" {
+			t.Fatalf("Recursive optical channel lookup failed for port: %s, parent of component %s not found.", p.Name(), compName)
+		}
+		compName = comp.GetParent()
+	}
+}
+
+func interfaceConfig(t *testing.T, dut *ondatra.DUTDevice, dp *ondatra.Port) {
```

Comment:
This function should also move to a shared library, probably at `internal/cfgplugins/interface.go`

Reply:
Perhaps `NewOCInterface` can be reused https://github.com/openconfig/featureprofiles/blob/2ef8bf6c3b9c6b911b97997559b21c3e1054e9a5/internal/attrs/attrs.go#L108-L111